### ISSUE TITLE
"Python config failure:" with no output

### ISSUE THREAD

**Issue by Lishen1** [pybind/pybind11#2127]
## Issue description
cmake failure with:
python-3.8.1.amd64
win 10 x64
 CMake Error at pybind11/tools/FindPythonLibsNew.cmake:96 (message):
Python config failure:

## Reproducible example code
command execution gives:
<string>:1: DeprecationWarning: SO is deprecated, use EXT_SUFFIX
just a warning but makes _PYTHON_SUCCESS non zero

**Comment by YannickJadoul** [comment]
What exactly did you run? Can we see the full log? Do you have a minimal reproducing example? Did you try cleaning out your build directory?

Too many unknowns here. If this is still an issue, please reopen the issue with more information!

**Comment by lukedan** [comment]
I'm having the same issue. I've been using the latest version of `pybind11` library provided by vcpkg on Windows 10, and I get the same error. Here's the full error log:
```
1> [CMake]   Python config failure:
1> [CMake] 
1> [CMake] Call Stack (most recent call first):
1> [CMake]   D:/Software/vcpkg/scripts/buildsystems/vcpkg.cmake:861 (_find_package)
1> [CMake]   D:/Software/vcpkg/installed/x64-windows/share/pybind11/pybind11Tools.cmake:38 (find_package)
1> [CMake]   D:/Software/vcpkg/installed/x64-windows/share/pybind11/pybind11Common.cmake:195 (include)
1> [CMake]   D:/Software/vcpkg/installed/x64-windows/share/pybind11/pybind11Config.cmake:249 (include)
1> [CMake]   D:/Software/vcpkg/scripts/buildsystems/vcpkg.cmake:861 (_find_package)
1> [CMake]   plugins/python_plugin_host_pybind11/CMakeLists.txt:12 (find_package)
1> [CMake] -- Configuring incomplete, errors occurred!
```
The CMakeOutput.log file does not seem to contain anything useful.

**Comment by lukedan** [comment]
Also, if I use a standalone installation of Python3 from Miniconda and add its path to `PYTHONHONE` and `PATH`, CMake generation finishes successfully.

**Comment by Einsied** [comment]
I encountered this behavior while compiling the package via conan under windows(pybind11  version 2.9.1). The challenge lies within the test package. More specific in the _python_interpreter method which returns "python" as path to the python interpreter. This works fine under most operating systems but not under windows. It should return the path to your python executable. After I added a special case for windows and a path to my python executable the problem was solved.
